staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/admission.go

```diff
@@ -0,0 +1,316 @@
+
+	return &MutatingWebhook{
+		Handler: admission.NewHandler(
+			admission.Connect,
+			admission.Create,
+			admission.Delete,
+			admission.Update,
+		),
+		clientManager: cm,
+	}, nil
+}
+
+// MutatingWebhook is an implementation of admission.Interface.
+type MutatingWebhook struct {
+	*admission.Handler
+	hookSource       WebhookSource
+	namespaceMatcher namespace.Matcher
+	clientManager    config.ClientManager
+	convertor        versioned.Convertor
+	jsonSerializer   runtime.Serializer
+}
+
+var (
+	_ = genericadmissioninit.WantsExternalKubeClientSet(&MutatingWebhook{})
+)
+
+// TODO find a better way wire this, but keep this pull small for now.
+func (a *MutatingWebhook) SetAuthenticationInfoResolverWrapper(wrapper config.AuthenticationInfoResolverWrapper) {
+	a.clientManager.SetAuthenticationInfoResolverWrapper(wrapper)
+}
+
+// SetServiceResolver sets a service resolver for the webhook admission plugin.
+// Passing a nil resolver does not have an effect, instead a default one will be used.
+func (a *MutatingWebhook) SetServiceResolver(sr config.ServiceResolver) {
+	a.clientManager.SetServiceResolver(sr)
+}
+
+// SetScheme sets a serializer(NegotiatedSerializer) which is derived from the scheme
+func (a *MutatingWebhook) SetScheme(scheme *runtime.Scheme) {
+	if scheme != nil {
+		a.clientManager.SetNegotiatedSerializer(serializer.NegotiatedSerializerWrapper(runtime.SerializerInfo{
+			Serializer: serializer.NewCodecFactory(scheme).LegacyCodec(admissionv1alpha1.SchemeGroupVersion),
+		}))
+		a.convertor.Scheme = scheme
+		a.jsonSerializer = json.NewSerializer(json.DefaultMetaFactory, scheme, scheme, false)
+	}
+}
+
+// WantsExternalKubeClientSet defines a function which sets external ClientSet for admission plugins that need it
+func (a *MutatingWebhook) SetExternalKubeClientSet(client clientset.Interface) {
+	a.namespaceMatcher.Client = client
+	a.hookSource = configuration.NewMutatingWebhookConfigurationManager(client.AdmissionregistrationV1alpha1().MutatingWebhookConfigurations())
+}
+
+// SetExternalKubeInformerFactory implements the WantsExternalKubeInformerFactory interface.
+func (a *MutatingWebhook) SetExternalKubeInformerFactory(f informers.SharedInformerFactory) {
+	namespaceInformer := f.Core().V1().Namespaces()
+	a.namespaceMatcher.NamespaceLister = namespaceInformer.Lister()
+	a.SetReadyFunc(namespaceInformer.Informer().HasSynced)
+}
+
+// ValidateInitialization implements the InitializationValidator interface.
+func (a *MutatingWebhook) ValidateInitialization() error {
+	if a.hookSource == nil {
+		return fmt.Errorf("the MutatingWebhook admission plugin requires a Kubernetes client to be provided")
+	}
+	if a.jsonSerializer == nil {
+		return fmt.Errorf("the MutatingWebhook admission plugin's jsonSerializer is not properly setup")
+	}
+	if err := a.namespaceMatcher.Validate(); err != nil {
+		return fmt.Errorf("the MutatingWebhook.namespaceMatcher is not properly setup: %v", err)
```

Comment:
-the

Reply:
Done here and in the validating webhook.